corgi/tasks/brew.py

```diff
@@ -442,15 +377,9 @@ def save_container(softwarebuild: SoftwareBuild, build_data: dict) -> ComponentN
             )
 
             set_license_declared_safely(obj, license_declared_raw)
-            image_arch_node, _ = ComponentNode.objects.get_or_create(
-                type=ComponentNode.ComponentNodeType.PROVIDES,
-                parent=root_node,
-                purl=obj.purl,
-                defaults={
-                    "object_id": obj.pk,
-                    "obj": obj,
-                },
-            )
+            image_arch_node = save_node(ComponentNode.ComponentNodeType.PROVIDES, root_node, obj)
+            # TODO: These are build-time dependencies, right?
+            #  Should we use PROVIDES_DEV here?
```

Comment:
This is partially true based on a conversation I had with the container factory team. All image components are build-time dependencies except for the ones from the base layer (ie UBI). That is because the other layers are part of multi-stage build and are discarded after the build. For example an image containing golang binaries often uses an intermediate layer to install the go compiler to build a go-binary, but that go compiler is no longer required at runtime, nor is it included in the final image.
I'm fine with leaving this as a TODO item for now, but perhaps update the comment with the extra information from this comment?